File: 189A/189A.cpp
#include<bits/stdc++.h>
using namespace std;
int main(){
    int n,a,b,c;
    int ans=0;
    cin >> n >> a >> b >> c;
    for(int i=0; i*a<=n; i++){
        for(int j=0; i*a+j*b<=n; j++){
            int k = (n - i*a - j*b)/c;
            if((i*a+j*b+k*c)==n){
                if(i+j+k>ans){
                    ans = max(i+j+k,ans);
                }     
            }
        }
    }
    cout << ans << endl;
    return 0;
} 

File: 189A/189A.py
s = [int(x) for x in input().split()]
ans = 0

for i in range(0, s[0]):
    for j in range(0, s[0]):
        tmp = s[0]-i-j
        if tmp < 0:
            continue
        if i % s[1] == 0 and j % s[2] == 0 and tmp % s[3] == 0:
            tmp2 = int(i/s[1] + j/s[2] + tmp/s[3])
            ans = max(ans, tmp2)
print(ans)
